### PR TITLE
Roles and permissions docs

### PR DESCRIPTION
**Pull request by rachelnabors**
This PR adds updates the "Organizations" section to be "Organizations, Roles, and Permissions":
image

It adds and/or makes significant changes to the following pages:
* [Organizations overview](https://docs-preview-514.clerkpreview.com/docs/organizations/overview)—conceptual overview
* [Roles and Permissions](https://docs-preview-514.clerkpreview.com/docs/organizations/roles-permissions)—conceptual overview
* [Create Roles and assign Permissions with Clerk](https://docs-preview-514.clerkpreview.com/docs/organizations/create-roles-permissions)—guide
* [Restrict access and authorize users with `protect` and `has` helpers](https://docs-preview-514.clerkpreview.com/docs/organizations/permissions-show-hide-content)—guide
* [Reassign the Creator Role](https://docs-preview-514.clerkpreview.com/docs/organizations/creator-role)—guide
* [`<Protect />`](https://docs-preview-514.clerkpreview.com/docs/components/protect)—reference

### CODE REVIEW

File: docs/manifest.json [hunk 1]
```diff
@@ -98,14 +98,19 @@
     ]
   ],
   [
-    { "title": "Organizations", "icon": "user-group", "root": "organizations" },
+    { "title": "Organizations, Roles, and Permisions", "icon": "user-group", "root": "organizations" },
```
Comment (kylemac) [inline docs/manifest.json:101]:
```suggestion
    { "title": "Organizations, Roles, and Permissions", "icon": "user-group", "root": "organizations" },
```

File: docs/manifest.json [hunk 2]
```diff
@@ -225,7 +225,7 @@
       ],
       ["<ClerkLoaded />", "/components/control/clerk-loaded"],
       ["<ClerkLoading />", "/components/control/clerk-loading"],
-      ["<Gate />", "/components/roles-permissions/gate"],
+      ["<Protect />", "/components/protect"],
```
Comment (panteliselef) [inline docs/manifest.json:228]: Should we do `/control/protect` to align with the rest ?

**Review by chanioxaris — APPROVED** [review]

File: docs/organizations/roles-permissions.mdx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+Custom Roles can be granted Permissions and access. For example you can create a new Role of “billing” (`org:billing`) which can be used to group users who belong to a specific department of the Organization and have permission to manage credit card information, invoices, and other resources related to billing.
+
+### The Creator Role
+
+When a user creates a new Organization, that user is automatically added as the Organization's first member and assigned the Creator Role. By default, Admin is the Creator Role.
+
+<Callout type="warning">
+You can't delete a Role if it's used as the Organization Creator Role. However, you can [reassign the Creator Role to another role](/docs/organizations/creator-role).
+</Callout>
```
Comment (JacobMGEvans) [inline docs/organizations/roles-permissions.mdx:37]: What happens when you delete a `Role` with users that isn't used as Creator Role?

Comment (rachelnabors) [reply docs/organizations/roles-permissions.mdx:37]: Good question. You can't. You can only delete Roles that have no assignees... so you'd need to migrate everyone to a new one, then delete it.

**Review by JacobMGEvans — APPROVED** [review]
Looks amazing. I left a question about deleting a Role, mainly because the deleting a Role with the `Creator` came up twice which made me think "what happens normally then?" which I didn't see mention of or missed it.

File: docs/organizations/roles-permissions.mdx (hunk 3 above)
Comment (JacobMGEvans) [reply docs/organizations/roles-permissions.mdx:37]: Ooooh Interesting! Thats what I was assuming based off what I read and the `Creator Role` information.

**Pull request merged**